internal/pkg/keeper/client.go

```diff
@@ -0,0 +1,353 @@
+)
+
+type keeperClient struct {
+	config              *types.Config
+	keeperUrl           string
+	serviceKey          string
+	serviceHost         string
+	servicePort         int
+	healthCheckRoute    string
+	healthCheckInterval string
+}
+
+// NewKeeperClient creates new Keeper Client. Service details are optional, not needed just for configuration, but required if registering
+func NewKeeperClient(registryConfig types.Config) (*keeperClient, error) {
+	client := keeperClient{
+		config:     &registryConfig,
+		serviceKey: registryConfig.ServiceKey,
+		keeperUrl:  registryConfig.GetRegistryUrl(),
+	}
+
+	// ServiceHost will be empty when client isn't registering the service
+	if registryConfig.ServiceHost != "" {
+		client.servicePort = registryConfig.ServicePort
+		client.serviceHost = registryConfig.ServiceHost
+		client.healthCheckRoute = registryConfig.CheckRoute
+		client.healthCheckInterval = registryConfig.CheckInterval
+	}
+
+	return &client, nil
+}
+
+// IsAlive simply checks if Keeper is up and running at the configured URL
+func (k *keeperClient) IsAlive() bool {
+	client := http.Client{Timeout: defaultTimeout}
+	resp, err := client.Get(k.keeperUrl + common.ApiPingRoute)
+	if err != nil {
+		return false
+	}
+
+	if resp.StatusCode >= http.StatusOK && resp.StatusCode < http.StatusMultipleChoices {
+		return true
+	}
+
+	return false
+}
+
+// Register registers the current service with Keeper for discovery and health check
+func (k *keeperClient) Register() error {
+	if k.serviceKey == "" || k.serviceHost == "" || k.servicePort == 0 ||
+		k.healthCheckRoute == "" || k.healthCheckInterval == "" {
+		return fmt.Errorf("unable to register service with keeper: Service information not set")
+	}
+
+	registrationReq := requests.AddRegistrationRequest{
+		BaseRequest: dtoCommon.BaseRequest{
+			Versionable: dtoCommon.Versionable{ApiVersion: common.ApiVersion},
+		},
+		Registration: dtos.Registration{
+			ServiceId: k.serviceKey,
+			Host:      k.serviceHost,
+			Port:      k.servicePort,
+			HealthCheck: dtos.HealthCheck{
+				Interval: k.healthCheckInterval,
+				Path:     k.healthCheckRoute,
+				Type:     "http",
+			},
+		},
+	}
+
+	jsonEncodedData, err := json.Marshal(registrationReq)
+	if err != nil {
+		return fmt.Errorf("failed to encode registration request: %v", err)
+	}
+
+	// check if the service registry exists first
+	resp, err := getRegistryByService(k.config.GetRegistryUrl() + ApiRegistrationByServiceIdRoute + k.serviceKey)
+	if err != nil {
+		return fmt.Errorf("failed to check the %s service registry status: %v", k.serviceKey, err)
+	}
+
+	// call the PUT registry API to update the registry if the service already exists
+	// otherwise, call the POST API to create the registry
+	httpMethod := http.MethodPost
+	if resp.StatusCode == http.StatusOK {
+		httpMethod = http.MethodPut
+	}
+	req, err := http.NewRequest(httpMethod, k.config.GetRegistryUrl()+common.ApiRegisterRoute, bytes.NewReader(jsonEncodedData))
```

Comment:
use the contract client instead of the http package call for all the methods
https://github.com/edgexfoundry/go-mod-core-contracts/blob/b0286c308de37bd3d0f6157ee2560331f20d5af8/clients/interfaces/registry.go#L17